dbtmetabase/metabase.py

```diff
@@ -404,14 +434,192 @@ def build_metadata_lookups(
 
         return table_lookup, field_lookup
 
+    def extract_exposures(
+        self,
+        models: List[MetabaseModel],
+        output_path: str = "./",
+        output_name: str = "metabase_exposures",
+        include_personal_collections: bool = True,
+        exclude_collections: Iterable = None,
+    ):
+
+        if exclude_collections is None:
+            exclude_collections = []
+
+        refable_models = {node.name: node.ref for node in models}
+
+        self.collections = self.api("get", "/api/collection")
+        self.tables = self.api("get", "/api/table")
+        self.table_map = {table["id"]: table["name"] for table in self.tables}
+
+        duplicate_name_check = []
+        captured_exposures = []
+
+        for collection in self.collections:
```

Comment:
This loop is very long and I'm struggling to read it. Would it make sense to break it up into a few functions? For example, body of each conditional branch in a separate function or something along those lines. If we measured cyclomatic complexity as part of linting, this function's would be through the roof.

Reply:
done